### PR TITLE
Replace `strpos()` and `mb_strpos()` with `str_starts_with()`

### PR DESCRIPTION
**Pull request by zonuexe**

### CODE REVIEW

File: lib/Completion/Bridge/TolerantParser/WorseReflection/WorseClassMemberCompletor.php [hunk 1]
```diff
@@ -81,7 +85,7 @@ public function complete(Node $node, TextDocument $source, ByteOffset $offset):
 
         foreach ($type->expandTypes()->classLike() as $type) {
             foreach ($this->populateSuggestions($nodeContext, $type, $static, $shouldCompleteOnlyName, $isInstance) as $suggestion) {
-                if ($partialMatch && 0 !== mb_strpos($suggestion->name(), $partialMatch)) {
+                if ($partialMatch && !str_starts_with($suggestion->name(), $partialMatch)) {
```
Comment (zonuexe) [inline lib/Completion/Bridge/TolerantParser/WorseReflection/WorseClassMemberCompletor.php:88]: Generally, don't need the `mb_` version of the function for this purpose. Look at <https://wiki.php.net/rfc/str_contains#case-insensitivity_and_multibyte_strings>.

File: lib/Indexer/Model/Query/Criteria/FileAbsolutePathBeginsWith.php [hunk 2]
```diff
@@ -18,7 +19,6 @@ public function isSatisfiedBy(Record $record): bool
             return false;
         }
 
-        $begins = strpos($record->filePath(), $this->prefix);
-        return $begins !== false && 0 === $begins;
+        return str_starts_with($record->filePath() ?? '', $this->prefix);
```
Comment (zonuexe) [inline lib/Indexer/Model/Query/Criteria/FileAbsolutePathBeginsWith.php:22]: Removing this `?? ''` will cause a runtime error.

**Comment by dantleech** [comment]
looks good! is it just a tidy up or is it related to a bug?

**Comment by zonuexe** [comment]
@dantleech Although I have not yet investigated the detailed conditions of occurrence, I found from the runtime error that there is a possibility of optimization here.

**Comment by dantleech** [comment]
thanks :slightly_smiling_face:

**Comment by mamazu** [comment]
This might have been an oversight on the Rector part and me, when migrating to php8. There might be more of those hidden in the code.

**Pull request merged**